openquake/server/dbserver.py

```diff
@@ -0,0 +1,127 @@
+from openquake.server.db import actions
+from openquake.server.settings import DATABASE
+from django.db import connection
+
+queue = Queue()
+
+
+def run_command(cmd, args, conn):
+    """
+    Execute the received command. Errors are trapped and a pair
+    (result, exctype) is sent back.
+    `exctype` is None if there is no exception, otherwise it is an exception
+    class and `result` is an error string containing the traceback.
+    """
+    try:
+        logging.info('Processing %s%s', cmd, args)
+        func = getattr(actions, cmd)
+
+        # execute the function by trapping any possible exception
+        res, etype, _ = safely_call(func, args)
+        if etype:
+            logging.error(res)
+
+        # send back the result and the exception class
+        conn.send((res, etype))
+    finally:
+        conn.close()
+
+
+def run_commands():
+    """
+    Execute the received commands in a queue.
+    """
+    connection.cursor()  # bind the db
+    while True:
+        conn, cmd, args = queue.get()
+        if cmd == 'stop':
+            conn.send((None, None))
+            conn.close()
+            break
+        run_command(cmd, args, conn)
+
+
+class DbServer(object):
+    """
+    A server collecting the received commands into a queue
+    """
+    def __init__(self, address, authkey):
+        self.address = address
+        self.authkey = authkey
+        self.thread = Thread(target=run_commands)
+
+    def loop(self):
+        listener = Listener(self.address, backlog=5, authkey=self.authkey)
+        logging.warn('DB server listening on %s:%d...' % self.address)
+        self.thread.start()
+        cmd = None
+        try:
+            while cmd != 'stop':
+                try:
+                    conn = listener.accept()
+                except KeyboardInterrupt:
+                    break
+                except:
+                    # unauthenticated connection, for instance by a port
+                    # scanner such as the one in manage.py
+                    continue
+                cmd_ = conn.recv()  # a tuple (name, arg1, ... argN)
+                cmd, args = cmd_[0], cmd_[1:]
+                if cmd.startswith('@'):  # slow command, run in process
+                    cmd = cmd[1:]  # strip @
+                    proc = Process(
+                        target=run_command, name=cmd, args=(cmd, args, conn))
+                    proc.start()
+                    logging.warn('Started %s%s in process %d',
+                                 cmd, args, proc.pid)
+                else:
+                    queue.put((conn, cmd, args))
+        finally:
+            listener.close()
+            self.thread.join()
+
+
+def runserver(hostport=None, loglevel='WARN'):
+    logging.basicConfig(level=getattr(logging, loglevel))
+    if hostport:  # assume a string of the form "hostname:port"
+        fname, port = hostport.split(':')
+        addr = (DATABASE['HOST'], int(port))
+        DATABASE['NAME'] = fname
```

Comment:
Please check the passed parameters (it seems you are mixing the hostname with the cfg file name)